Assets/Scripts/Mixed/Systems/KinematicCharacterControllerSystems.cs

```diff
@@ -163,18 +166,26 @@ public class KCCJumpSystem : SystemBase
     {
         protected override void OnUpdate()
         {
+            float deltaTime = Time.DeltaTime;
             Entities.ForEach((
                 ref KCCVelocity velocity,
-                in KCCJumping jumping,
+                ref KCCJumping jumping,
                 in KCCGrounded grounded,
                 in KCCGravity gravity) =>
                 {
                     // If the KCC is attempting to jump and is grounded, jump
-                    if (jumping.attemptingJump && !grounded.Falling)
+                    if (jumping.attemptingJump && grounded.elapsedFallTime <= jumping.jumpGraceTime && jumping.timeElapsedSinceJump >= jumping.jumpCooldown)
                     {
                         velocity.worldVelocity += gravity.Up * jumping.jumpForce;
+                        jumping.timeElapsedSinceJump = 0.0f;
                     }
                     // Otherwise, do nothing
+
+                    // Track jumping cooldown
+                    if (jumping.timeElapsedSinceJump < jumping.jumpCooldown)
+                    {
+                        jumping.timeElapsedSinceJump += deltaTime;
+                    }
```

Comment:
Make this an else if clause, we don't want to increment time for the frame the player is jumping :)

Reply:
Alright, will do!